### PR TITLE
sapmon will now also accept keyvault url and msi client id instead of password

### PR DESCRIPTION
**Pull request by PakDLiu**
The sapmon.py script will now also accept KeyVault URL and MSI Client ID to get the password.

Testing manually on my own collector VM, no errors. Saw password get fetched correctly

### CODE REVIEW

**Comment by pabowers** [comment]
Why is this PR affecting the Pipeline build check? It seems a little unrelated

**Comment by tniek** [comment]
> Why is this PR affecting the Pipeline build check? It seems a little unrelated

Because PR #182 has not been merged yet.

**Comment by tniek** [comment]
> Looks good to me, just look into why this is causing the pipeline to fail

Bad branch name (@PakDLiu - I made the same mistake, check the last sentence in our [Contributing Guidelines](https://github.com/Azure/sap-hana/blob/master/CONTRIBUTING.md) :)) causes the VNET creation of the (unrelated) auto-deployment to fail.

File: monitor/sapmon.py [hunk 1]
```diff
@@ -32,27 +33,37 @@ class SapHana:
    connection = None
    cursor     = None
 
-   def __init__(self, host = None, port = None, user = None, password = None, hanaDetails = None):
+   def __init__(self, host = None, port = None, user = None, password = None, passwordKeyVault = None, msiClientId = None, hanaDetails = None):
```
Comment (tniek) [inline monitor/sapmon.py]: I would actually prefer to rename `passwordKeyVault` into `keyVaultUrl`, as this describes more adequately what it actually is.

File: monitor/sapmon.py [hunk 2]
```diff
@@ -203,13 +214,13 @@ def getComputeInstance(operation):
          )["compute"]
 
    @staticmethod
-   def getAuthToken(resource):
+   def getAuthToken(resource, clientId = None):
```
Comment (tniek) [inline monitor/sapmon.py]: For consistency (line 38), I would recommend renaming `clientId` into `msiClientId`.

File: monitor/sapmon.py [hunk 3]
```diff
@@ -220,9 +231,9 @@ class AzureKeyVault:
    """
    params  = {"api-version": "7.0"}
 
-   def __init__(self, keyvaultName):
+   def __init__(self, keyvaultName, clientId = None):
```
Comment (tniek) [inline monitor/sapmon.py]: See above: For consistency (line 38), I would recommend renaming `clientId` into `msiClientId`.

**Comment by persiaAziz** [comment]
Thanks for this PR Donald, this PR tests PR 182 ✔️

File: monitor/sapmon.py (hunk 1 above)
Comment (tniek) [reply monitor/sapmon.py]: See my review comment: It actually makes more sense to remove the entire KeyVault business from the `Hana` class and move it into `_Context`, presumably there into `parseSecrets()`

**Comment by PakDLiu** [comment]
Note:
If the VM has more than one user assigned identities assigned to it, the client ID is required.

Update:
If we are fetching the password from the customer's keyvault (IE: multiple identities assigned to the VM), this script will check the tags for the sapmon MSI client id.

There is not changes in the current flow, since it will try to find the tag, won't find it, and fetch the MSI token with no client id (will work since there is only 1 MSI assigned to it), and everything will work as normal.

**Pull request merged**